onnxruntime/core/providers/qnn/builder/opbuilder/gather_op_builder.cc

```diff
@@ -118,6 +177,45 @@ Status GatherOpBuilder::ProcessInputs(QnnModelWrapper& qnn_model_wrapper,
   return Status::OK();
 }
 
+Status GatherOpBuilder::ProcessInputs(QnnModelWrapper& qnn_model_wrapper,
+                                      const NodeUnit& node_unit,
+                                      const logging::Logger& logger,
+                                      bool is_quantized_model,
+                                      std::vector<std::string>& input_names,
+                                      bool do_op_validation) const {
+  const auto& inputs = node_unit.Inputs();
+  ORT_RETURN_IF(inputs.size() != 2, "QNN EP: Gather operator must have two inputs");
+  ORT_RETURN_IF_ERROR(ProcessInput(qnn_model_wrapper, inputs[0], logger, is_quantized_model, input_names));
+
+  return ProcessIndicesInput(qnn_model_wrapper, inputs[1], true, logger, is_quantized_model, input_names, do_op_validation);
```

Comment:
how to determine int32_type_is_signed? Why use singed int32 for Gather, and unsigned for GatherND and GatherElements?

Reply:
For GatherND, QNN prints the warning `[WARNING] Input[1] has Datatype 0x32` when the type is signed.
For GatherElements, either type could work, but negative indices are not supported in general for any Gather op.

In any case, I just left a comment explaining why I made this back into a draft. There are too many inconsistencies between QNN Gather* ops and ONNX Gather* ops.